### DISCUSSION TITLE
Dependencies for running in .net framework 4.7.2?

### DISCUSSION THREAD

**Discussion by Gobling** [QuestPDF/QuestPDF#982] [Q&A] [answered]
Hi

I've build a nice report generator using QuestPDF in a .NET 8.0 console application, which I'm trying to transfer to a web server in a project running .Net Framework 4.7.2, but I'm getting the following error message:

> Could not load file or assembly 'QuestPDF, Version=2024.7.3.0, Culture=neutral, PublicKeyToken=null' or one of its dependencies. The system cannot find the file specified.

I took all the code out, so all that is left is: 

> QuestPDF.Settings.License = LicenseType.Community;

And it's still failing. 

I installed QuestPDF using Nuget in visual studio. The path reference is: C:\....\packages\QuestPDF.2024.7.3\lib\netstandard2.0\QuestPDF.dll and I've checked that the file exists. 

I have multiple other packages installed the same way in the project, but this one is challenging. Is there a missing dependency? Any other ideas?

**Comment by nomadstays** [comment]
did you try copying the QuestPDF.dll into the /bin folder and adding      QuestPDF.Settings.License = LicenseType.Community;
 into your _Appstart.cshtml file?
 
 I note this technique doesn't work with the latest 2025.4.0 version.

**Reply by AyoubWalid** [reply]
I spent all day trying to fix this, you will need to copy the QuestPdfSkia.dll into Windows\System32 and Windows\SysWOW64 manually to make it work.

**Reply by Rbcody** [reply]
Thank you!! It seems that many people are encountering this dependency issue. Your answer solved the problem for me.

**Reply by Mucip** [reply]
Hi,
Yes. I spend many hours and found answer finally. ;)

Regards,
Mucip:)

**Comment by DawidRobertNowak** [comment, marked as answer]
Hello,

We are closing this thread due to inactivity. We apologize for the delayed response. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

This error occurs because QuestPDF's native rendering dependency is not being copied to the application's output directory. On .NET Framework projects this often does not happen automatically, since the native assets shipped under the package's `runtimes` folder are not always restored as they are on modern .NET. The recommended approach is to ensure the native library is published alongside your application, for example by targeting an explicit platform (such as `x64`) so the matching native runtime assets are included in the build output. We would not recommend copying the native library into system-wide Windows directories, as that is not deployable and can affect other applications.

If you are still experiencing this on a current version, please open a new issue with your project type and configuration so we can take a closer look.

Thank you!

**Comment by nomadstays** [comment]
Its worth remembering that .net framework is still a supported release product by Microsoft.  Maybe ensuring the copy function works for all supported versions of QuestPDF would be the answer.

**Reply by MarcinZiabek** [reply]
Fair point! Although we have made several improvements to increase compatibility with older Windows environments and .NET Framework targets, we plan to revisit this further over the coming months. QuestPDF already employs some fallback approaches (including copying the appropriate native files, and searching/detecting the most likely source and target locations where such files should be placed). Given that legacy .NET Framework experiences problems with native dependencies, some setups use truly uncommon configurations, and that security software may block certain fallback operations, it is always a matter of balance.

**Comment by Rbcody** [comment]
For what it’s worth, QuestPDF has made reporting MUCH easier for my software, even with .NET dependencies.  I previously programmed reporting by using Word DLL’s, and I still support them for some functions.  However, programming the Microsoft Word DLL’s is a mess!  QuestPDF is logical, reliable, and efficient, and it is a pleasure to use!
,
